src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCreativeTank.java

```diff
@@ -0,0 +1,151 @@
+        Textures.PIPE_OUT_OVERLAY.renderSided(this.getFrontFacing(), renderState, translation, pipeline);
+        Textures.FLUID_OUTPUT_OVERLAY.renderSided(this.getFrontFacing(), renderState, translation, pipeline);
+    }
+
+    @Override
+    public MetaTileEntity createMetaTileEntity(MetaTileEntityHolder holder) {
+        return new MetaTileEntityCreativeTank(this.metaTileEntityId);
+    }
+
+    @Override
+    protected ModularUI createUI(EntityPlayer entityPlayer) {
+        ModularUI.Builder builder = ModularUI.builder(GuiTextures.BACKGROUND, 176, 209)
+                .bindPlayerInventory(entityPlayer.inventory, 126);
+        builder.widget(new PhantomFluidWidget(36, 6, 18, 18,
+                () -> this.fluidTank.getFluid(), data -> {
+            this.fluidTank.setFluid(data);
+        })
+                .showTip(false).setBackgroundTexture(GuiTextures.SLOT_DARKENED));
+        builder.label(7, 9, "Fluid");
+        builder.widget(new ImageWidget(7, 45, 154, 14, GuiTextures.DISPLAY));
+        builder.widget(new TextFieldWidget2(9, 47, 152, 10, () -> String.valueOf(mBPerCycle), value -> {
+            if (!value.isEmpty()) {
+                mBPerCycle = Integer.parseInt(value);
+            }
+        }).setAllowedChars("0123456789").setMaxLength(19).setValidator(getTextFieldValidator()));
+        builder.label(7, 28, "mB per cycle");
+
+        builder.widget(new ImageWidget(7, 82, 154, 14, GuiTextures.DISPLAY));
+        builder.widget(new TextFieldWidget2(9, 84, 152, 10, () -> String.valueOf(ticksPerCycle), value -> {
+            if (!value.isEmpty()) {
+                ticksPerCycle = Integer.parseInt(value);
+            }
+        }).setMaxLength(10).setNumbersOnly(0, Integer.MAX_VALUE));
+        builder.label(7, 65, "Ticks per cycle");
+
+
+        builder.widget(new CycleButtonWidget(7, 101, 162, 20, () -> active, value -> active = value, "Not active", "Active"));
+
+        return builder.build(getHolder(), entityPlayer);
+    }
+
+    @Override
+    public void update() {
+        super.update();
+        if (getOffsetTimer() % ticksPerCycle != 0 || fluidTank.getFluid() == null
+                || getWorld().isRemote || !active) return;
+
+        FluidStack stack = fluidTank.getFluid().copy();
+
+        TileEntity tile = getWorld().getTileEntity(getPos().offset(this.getFrontFacing()));
+        if (tile != null) {
+            IFluidHandler fluidHandler = tile.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, frontFacing);
+            if (fluidHandler == null || fluidHandler.getTankProperties().length == 0)
+                return;
+
+            stack.amount = mBPerCycle;
+            int canInsertAmount = fluidHandler.fill(stack, false);
+            stack.amount = Math.min(mBPerCycle, canInsertAmount);
+
+            fluidHandler.fill(stack, true);
+        }
+    }
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound data) {
```

Comment:
These read and write methods should call super first, then save the information into data, and then return data